airbyte-integrations/connectors/destination-langchain/destination_langchain/document_processor.py

```diff
@@ -84,7 +84,7 @@ def _extract_metadata(self, record: AirbyteRecordMessage) -> dict:
         metadata[METADATA_STREAM_FIELD] = stream_identifier
         # if the sync mode is deduping, use the primary key to upsert existing records instead of appending new ones
         if current_stream.primary_key and current_stream.destination_sync_mode == DestinationSyncMode.append_dedup:
-            metadata[METADATA_RECORD_ID_FIELD] = self._extract_primary_key(record, current_stream)
+            metadata[METADATA_RECORD_ID_FIELD] = f"{stream_identifier}_{self._extract_primary_key(record, current_stream)}"
```

Comment:
Since I'm still not familiar with what's in the cdk - is this piece something that the cdk handles for you? (setting metadata cols / deduping rows)